src/test/kotlin/app/ControlTest.kt

```diff
@@ -17,7 +17,7 @@ class ControlTest {
         expectThat(readResource("app/${testControl.resourceName}.json", Control::class)) {
             get { this::class }.isEqualTo(testControl.type)
             get { name }.isEqualTo(testControl.controlName)
-            get { states.size }.isEqualTo(testControl.statesSize)
+            get { states?.size }.isEqualTo(testControl.statesSize)
```

Comment:
this is exactly the result of missing `@NotNull` :( - could you please investigate how to keep it?

Reply:
The geStates() method already had a @Nullable annotation, but i think the reason it now gives an error without the ? is because the @Nullable is now also on the field. Could that assumption be correct?

Reply:
Hah that was funny mistake - since nobody realize that here kotlin actualy uses a field since it has protected visiblity. Then OK it makes sense like this